utils/build-script-impl

```diff
@@ -222,6 +221,7 @@ KNOWN_SETTINGS=(
     cross-compile-hosts                           ""                "space-separated list of targets to cross-compile host Swift tools for"
     cross-compile-with-host-tools                 ""                "set to use the clang we build for the host to then build the cross-compile hosts"
     cross-compile-install-prefixes                ""                "semicolon-separated list of install prefixes to use for the cross-compiled hosts. The list expands, so if there are more cross-compile hosts than prefixes, unmatched hosts use the last prefix in the list"
+    cross-compile-deps-path                       ""                "path for CMake to look for cross-compiled library dependencies, such as libXML2"
```

Comment:
I thought about making this a list like `cross-compile-hosts` but realistically only one cross-compilation host is ever passed in, so this will do for now. The issue is that each cross-compilation host will require its own separate C/C++ cross-compilation toolchain, which there's no way to pass in to `build-script` yet. We're able to handle a single cross-compilation host with hacked-in flags like `--android-ndk` and `--native-clang-tools-path`, but for example, we can't even build multiple Android architectures with a single `build-script` invocation yet.

Reply:
@buttaface I think the last sentence is incorrect. I imagine that if one hacked in something similar to what we have on Darwin in the stdlib cmake (which is horrible btw and I am not suggesting this).

That being said, shouldn't this just be found in some notion of an SDK? I think that is how on Darwin this would work.

Reply:
The reason I said "we can't even build multiple Android architectures with a single build-script invocation yet" in my last sentence is that, for example, [we pass in the path to a particular libicu built for one arch when invoking `build-script`](https://github.com/apple/swift/pull/33724#issuecomment-736656884). Even on Darwin, only compiling the Swift compiler itself for macOS x86_64 was supported until recently, when support for cross-compiling it from macOS x86_64 to arm64 was added but not the other way, apple/swift-package-manager#3500, but again only a single cross-compilation host is supported on Darwin. Of course, one could hack in further changes to support multiple hosts, but it is currently not supported.

As for this just being an SDK, it is more general than that. I use this flag below to find dependencies like libxml2 or libsqlite3,  which are additional package dependencies that aren't usually included in cross-compilation SDKs. I found this useful when cross-compiling the toolchain for Android, [as I pass in the Termux prefix with those prebuilt library packages installed to this flag](https://github.com/termux/termux-packages/blob/c7626ec952b10f8a0ed0e8df0a204abc05807b7b/packages/swift/build.sh#L140) and the separate Android NDK, that has the libc sysroot but no libraries like libxml2, to the `--android-ndk` flag above.

This gives the most flexibility because if they are all in one SDK, you can always pass the same SDK path in to both flags.